### DISCUSSION TITLE
feature request: add `defaultPriorityFee` on chains

### DISCUSSION THREAD

**Discussion by roninjin10** [wevm/viem#968] [Idea / Feature Request]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.5.0

### Issue

The base priority fee for all chains defaults to [1.5 gwei](https://github.com/wagmi-dev/viem/blob/ed779e9d5667704fd7cfc5a2af032f7a2c3cae9e/src/utils/transaction/prepareRequest.ts#L44). 

For OP a better default would be `.001 gwei`

### Impact

This is a really good default for Ethereum but when it comes to OP this results in an extremely high percentage of users overpaying needlessly.

- Example (but with ethers): https://github.com/ethereum-optimism/developer-support/issues/27
- In our own analysis we have seen a huge % of transactions are overpaying

<img width="902" alt="image" src="https://github.com/wagmi-dev/viem/assets/35039927/fe7604c0-eb92-4f28-8b6e-cecd3ae4d1ad">

### Proposed fix

Wanted to open issue before I went ahead and did this fix but this is proposed fix
- Add `defaultPriorityFee` to chains object in the constants package
- Change the defaultPriorityFee to `chain.defaultPriorityFee ?? '1.5' 
- (out of scope) consider making sure all chains explicitly state their priority fee by making it a required property instead of optional

### Expected Behavior

Should default to much lower priority fee for OP chains like base and OP mainnet

### Steps To Reproduce

Create a transaction purely with wagmi defaults and send and priority fee will be set to 1.5

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://github.com/ethereum-optimism/developer-support/issues/27

### Anything else?

Happy to contribute and fix this one next Monday

**Comment by jxom** [comment]
Agree that `defaultPriorityFee` could be a nice addition! 

Fee calculations are not estimated for JSON-RPC Accounts  in `sendTransaction`/`writeContract` in viem as we defer fee calculations to the Wallet which handle 1559 fee data properly. 

In viem, fee calculations are only done for Local Accounts (ie. Private Key, Mnemonic accounts), and the main consumers of Local Accounts are: Wallets, scripts, etc. So this issue probably will affect those consumers instead (maybe Wallets less so, as they have more advanced 1559 fee calculations), so agree that `defaultPriorityFee` on the chain does make sense. 

Going to move this to a discussion for now as it is more a feature request, but happy to take a PR for this.
